PWGHF/HFL/Tasks/taskHFmuCharmBeautySep.cxx

```diff
@@ -0,0 +1,103 @@
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+///
+/// \Task to estimate HF->mu in the forward direction and use the DCA observable to separate b-> mu, c-> mu.
+/// \Command to run on terminal : o2-analysis-hfmu-charm-beauty-separation | o2-analysis-fwdtrackextension --aod-file AO2D.root
+/// \Author : Shreyasi Acharya <shreyasi.acharya@cern.ch>, LPC, France
+
+#include "Framework/runDataProcessing.h"
+#include "Framework/AnalysisTask.h"
+#include "TDatabasePDG.h"
+#include "PWGDQ/Core/VarManager.h"
+#include "CCDB/BasicCCDBManager.h"
+#include "Framework/AnalysisDataModel.h"
+#include "Framework/ASoAHelpers.h"
+#include "PWGDQ/DataModel/ReducedInfoTables.h"
+#include "PWGDQ/Core/HistogramManager.h"
+#include "PWGDQ/Core/MixingHandler.h"
+#include "PWGDQ/Core/AnalysisCut.h"
+#include "PWGDQ/Core/AnalysisCompositeCut.h"
+#include "ReconstructionDataFormats/TrackFwd.h"
+#include "Common/DataModel/TrackSelectionTables.h"
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+
+// using MyMuons = soa::Join<aod::FwdTracks, aod::FwdTracksDCA>;
+// Iterate on muon using the collision iterator in the dq-analysis style
+struct IterateFwdTracksMuons {
+
+  HistogramRegistry spectra{"ForwardTracks", {}, OutputObjHandlingPolicy::AnalysisObject, true, true};
+  void init(o2::framework::InitContext&)
+  {
+    AxisSpec trackTypeAxis = {6, -0.5, 5.5, "Track Type"};
+    AxisSpec ptRecoAxis = {1500, 0, 15, "#it{p}_{T}_{Reco}"};
+    AxisSpec dcaxAxis = {1000, -5.0, 5.0, "DCA {x or y} (cm)"};
+    AxisSpec etaRecoAxis = {150, -5, -2, "#eta_{Reco}"};
+    AxisSpec rAbsAxis = {100, 0, 100, "R_{abs}"};
+    AxisSpec pdcaAxis = {450, 0, 450, "p_{DCA}"};
+    AxisSpec chi2GlobalAxis = {170, -1.5, 150.5, "#chi^{2} global"};
+    AxisSpec chi2MCHMFTAxis = {170, -1.5, 150.5, "#chi^{2} MCH-MFT"};
+    AxisSpec chi2MCHMIDAxis = {170, -1.5, 150.5, "#chi^{2} MCH-MID"};
+
+    HistogramConfigSpec HistVariable({HistType::kTHnSparseF, {ptRecoAxis, dcaxAxis, etaRecoAxis, chi2MCHMFTAxis, chi2GlobalAxis, chi2MCHMIDAxis, rAbsAxis, pdcaAxis}});
+    spectra.add("hBasicDist", "", HistVariable);
+    HistogramConfigSpec HistTrackType({HistType::kTH1F, {trackTypeAxis}});
+    spectra.add("hTrackType", "", HistTrackType);
+
+    spectra.add("hDCAxMuonType0", " dcax", {HistType::kTH1F, {{1000, -5.0, 5.0}}});
+    spectra.add("hDCAyMuonType0", " dcay", {HistType::kTH1F, {{1000, -5.0, 5.0}}});
+    spectra.add("hDCAxyMuonType0", " dcaxy", {HistType::kTH1F, {{1000, 0.0, 10.0}}});
+  }
+  void process(aod::Collisions::iterator const& collision, soa::Join<aod::FwdTracks, aod::FwdTracksDCA> const& muons)
```

Comment:
```suggestion
  void process(aod::Collisions::iterator const& collision, soa::Join<aod::FwdTracks, aod::FwdTracksDCA> const& tracks)
```
These are not muons.

Reply:
In this tutorial code, "muons" is used as the forward tracks. In MC, these tracks have PID 13. Hence, I had used muons. I will change to tracks.
/Tutorials/src/muonIteration.cxx